driver-mapping/src/main/java/com/datastax/driver/mapping/configuration/naming/NamingConvention.java

```diff
@@ -0,0 +1,88 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.configuration.naming;
+
+import java.util.List;
+
+/**
+ * Represent a naming convention (e.g. snake_case, camelCase, etc...) to be used when
+ * auto-translating java property names to cassandra column names and vise-versa.
+ *
+ * This interface may be implemented to define custom naming convention.
+ */
+public interface NamingConvention {
+
+    /**
+     * Receive a property name value and returns an ordered list of Word objects.
+     * Each word contains a String value and a boolean indicating whether or not
+     * the value is an abbreviation (In most cases could not be determined).
+     * Quick examples:
+     * <ul>
+     * <li>Let's consider lowerCamelCase convention and input = "myXMLParser",
+     * then the output should be:
+     * [
+     *   word{value = "my", isAbbreviation = false},
+     *   word{value = "xml", isAbbreviation = true},
+     *   word{value = "parser", isAbbreviation = false}
+     * ]</li>
+     * <li>Let's consider lower_snake_case convention and input = "myXMLParser",
+     * then the output may be (since there's no trivial way to determine xml
+     * to an abbreviation):
+     * [
+     *   word{value = "my", isAbbreviation = false},
+     *   word{value = "xml", isAbbreviation = false},
+     *   word{value = "parser", isAbbreviation = false}
+     * ]</li>
+     * </ul>
+     *
+     * @param input value to split
+     * @return an ordered list of split Word objects
+     */
+    List<Word> split(String input);
+
+    /**
+     * Receive an ordered list of Word objects and returns a result property name.
+     * Quick examples:
+     * <ul>
+     * <li>Let's consider lowerCamelCase convention with upperCaseAbbreviations set
+     * to false, and input = [
+     *   word{value = "my", isAbbreviation = false},
+     *   word{value = "xml", isAbbreviation = true},
+     *   word{value = "parser", isAbbreviation = false}
+     * ]
+     * then the output should be "myXmlParser".</li>
+     * <li>Let's consider upperCamelCase convention with upperCaseAbbreviations set
+     * to true, and input = [
+     *   word{value = "my", isAbbreviation = false},
+     *   word{value = "xml", isAbbreviation = true},
+     *   word{value = "parser", isAbbreviation = false}
+     * ]
+     * then the output should be "MyXMLParser".</li>
+     * </ul>
+     *
+     * @param input list to translate
+     * @return the result property name
+     */
+    String join(List<Word> input);
+
+    /**
+     * @param other NamingConvention to test
+     * @return whether or not the given naming convention returns exactly
+     * the same output as this
+     */
+    boolean equals(NamingConvention other);
```

Comment:
It seems that this is only used to optimize the conversions in `NamingStrategy` when the two conventions are the same.
We could instead have a `PassThroughNamingStrategy` that contains `null` conventions and overrides `toCassandra` (and `toJava` if we keep it). It could even be a unique instance of an anonymous class exposed as a singleton, such as `NamingStrategy.PASS_THROUGH`.
This way we get rid of `equals` in all the implementations.